{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Table of Contents\n",
    " <p><div class=\"lev1\"><a href=\"#Task-1.-Compiling-Ebola-Data\"><span class=\"toc-item-num\">Task 1.&nbsp;&nbsp;</span>Compiling Ebola Data</a></div>\n",
    " <div class=\"lev1\"><a href=\"#Task-2.-RNA-Sequences\"><span class=\"toc-item-num\">Task 2.&nbsp;&nbsp;</span>RNA Sequences</a></div>\n",
    " <div class=\"lev1\"><a href=\"#Task-3.-Class-War-in-Titanic\"><span class=\"toc-item-num\">Task 3.&nbsp;&nbsp;</span>Class War in Titanic</a></div></p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import glob\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "pd.options.mode.chained_assignment = None  # default='warn', Mutes warnings when copying a slice from a DataFrame.\n",
    "\n",
    "PATH_FOLDER = \"../../Arsalan/Homework 1/Data\"\n",
    "PATH_GUINEA = PATH_FOLDER + \"/ebola/guinea_data\"\n",
    "PATH_LIBERIA = PATH_FOLDER + \"/ebola/liberia_data\"\n",
    "PATH_SL = PATH_FOLDER + \"/ebola/sl_data\"\n",
    "\n",
    "PATH_MICROBIOME = PATH_FOLDER + \"/microbiome\"\n",
    "\n",
    "na_val=['?', 'NaN', 'NA', 'Na', 'na', 'nan', 'Nan']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 1. Compiling Ebola Data\n",
    "\n",
    "The `DATA_FOLDER/ebola` folder contains summarized reports of Ebola cases from three countries (Guinea, Liberia and Sierra Leone) during the recent outbreak of the disease in West Africa. For each country, there are daily reports that contain various information about the outbreak in several cities in each country.\n",
    "\n",
    "Use pandas to import these data files into a single `Dataframe`.\n",
    "Using this `DataFrame`, calculate for *each country*, the *daily average per month* of *new cases* and *deaths*.\n",
    "Make sure you handle all the different expressions for *new cases* and *deaths* that are used in the reports."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Import all files and put them into df\n",
    "\n",
    "files_guinea = glob.glob(PATH_GUINEA + \"/*.csv\")\n",
    "data_guinea_array = [pd.read_csv(f, na_values=na_val) for f in files_guinea]\n",
    "data_guinea = pd.concat(data_guinea_array, ignore_index=True)\n",
    "data_guinea['Country'] = \"Guinea\"\n",
    "data_guinea['Year'] = [[int(s) for s in val.split('-') if s.isdigit()][0] for val in data_guinea['Date']]\n",
    "data_guinea['Month'] = [[int(s) for s in val.split('-') if s.isdigit()][1] for val in data_guinea['Date']]\n",
    "data_guinea['Day'] = [[int(s) for s in val.split('-') if s.isdigit()][2] for val in data_guinea['Date']]\n",
    "\n",
    "files_liberia = glob.glob(PATH_LIBERIA + \"/*.csv\")\n",
    "data_liberia_array = [pd.read_csv(f, na_values=na_val) for f in files_liberia]\n",
    "data_liberia = pd.concat(data_liberia_array, ignore_index=True)\n",
    "data_liberia['Country'] = \"Liberia\"\n",
    "data_liberia['Year'] = [[int(s) for s in val.split('/') if s.isdigit()][2] for val in data_liberia['Date']]\n",
    "data_liberia['Month'] = [[int(s) for s in val.split('/') if s.isdigit()][0] for val in data_liberia['Date']]\n",
    "data_liberia['Day'] = [[int(s) for s in val.split('/') if s.isdigit()][1] for val in data_liberia['Date']]\n",
    "\n",
    "files_sl = glob.glob(PATH_SL + \"/*.csv\")\n",
    "data_sl_array = [pd.read_csv(f, na_values=na_val) for f in files_sl]\n",
    "data_sl = pd.concat(data_sl_array, ignore_index=True)\n",
    "data_sl['Country'] = \"SierraLeone\"\n",
    "data_sl['Year'] = [[int(s) for s in val.split('-') if s.isdigit()][0] for val in data_sl['date']]\n",
    "data_sl['Month'] = [[int(s) for s in val.split('-') if s.isdigit()][1] for val in data_sl['date']]\n",
    "data_sl['Day'] = [[int(s) for s in val.split('-') if s.isdigit()][2] for val in data_sl['date']]\n",
    "\n",
    "dfList = [data_guinea, data_liberia, data_sl]\n",
    "df = pd.concat(dfList, ignore_index=True)\n",
    "\n",
    "\n",
    "# Cleaning the data\n",
    "\n",
    "def mergeColumns(data_frame, column_to_fill, columns_to_empty):\n",
    "    for column in columns_to_empty:\n",
    "        data_frame[column_to_fill] = data_frame[column_to_fill].fillna(df[column])\n",
    "        \n",
    "mergeColumns(df, 'Description', ['variable', 'Variable'])\n",
    "mergeColumns(df, 'National', ['Totals'])\n",
    "mergeColumns(df, 'Police training School', ['Police traning School'])\n",
    "df = df.drop('variable', axis=1)\n",
    "df = df.drop('Variable', axis=1)\n",
    "df = df.drop('Totals', axis=1)\n",
    "df = df.drop('Date', axis=1)\n",
    "df = df.drop('date', axis=1)\n",
    "df = df.drop('Police traning School', axis=1)\n",
    "\n",
    "cities = list(set(df.columns)-set(['Country', 'Day', 'Month', 'Year', 'Description', 'National']))\n",
    "\n",
    "df=df.fillna(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def avgOverMonth(country, description, month):\n",
    "    value = 0\n",
    "    if (type(description) == str):\n",
    "        value = df.query('Description == @description and Month == @month and Country == @country')['National'].astype(dtype=int).mean()\n",
    "    elif (type(description) == list):\n",
    "        for desc in description:\n",
    "            value += avgOverMonth(country, desc, month)\n",
    "    return value\n",
    "\n",
    "def computeAvgNewDeaths(country, description, month):\n",
    "    firstDay = df.query('Country == @country and Month == @month')['Day'].min()\n",
    "    lastDay = df.query('Country == @country and Month == @month')['Day'].max()\n",
    "    if (country == 'SierraLeone' and month == 12):\n",
    "        lastDay = 6\n",
    "    firstDayValue = 0\n",
    "    lastDayValue = 0\n",
    "    if (type(description) == str):\n",
    "        firstDayValue = df.query('Description == @description and Day == @firstDay and Country == @country and Month == @month')['National'].astype(dtype=int)[df.query('Description == @description and Day == @firstDay and Country == @country and Month == @month')['National'].index[0]]\n",
    "        lastDayValue = df.query('Description == @description and Day == @lastDay and Country == @country and Month == @month')['National'].astype(dtype=int)[df.query('Description == @description and Day == @lastDay and Country == @country and Month == @month')['National'].index[0]]\n",
    "    elif (type(description) == list):\n",
    "        for desc in description:\n",
    "            firstDayValue += df.query('Description == @desc and Day == @firstDay and Country == @country and Month == @month')['National'].astype(dtype=int)[df.query('Description == @desc and Day == @firstDay and Country == @country and Month == @month')['National'].index[0]]\n",
    "            lastDayValue += df.query('Description == @desc and Day == @lastDay and Country == @country and Month == @month')['National'].astype(dtype=int)[df.query('Description == @desc and Day == @lastDay and Country == @country and Month == @month')['National'].index[0]]\n",
    "    return (lastDayValue - firstDayValue)/(lastDay - firstDay)\n",
    "    \n",
    "daily_average_deaths_Guinea_August = avgOverMonth('Guinea', 'New deaths registered today', 8)\n",
    "daily_average_deaths_Guinea_September = avgOverMonth('Guinea', 'New deaths registered', 9)\n",
    "daily_average_deaths_Guinea_October = avgOverMonth('Guinea', 'New deaths registered', 10)\n",
    "\n",
    "daily_average_newCases_Guinea_August = avgOverMonth('Guinea', 'Total new cases registered so far', 8)\n",
    "daily_average_newCases_Guinea_September = avgOverMonth('Guinea', 'Total new cases registered so far', 9)\n",
    "daily_average_newCases_Guinea_October = avgOverMonth('Guinea', 'Total new cases registered so far', 10)\n",
    "\n",
    "\n",
    "daily_average_deaths_Liberia_June = avgOverMonth('Liberia', 'Newly reported deaths', 6)\n",
    "daily_average_deaths_Liberia_July = avgOverMonth('Liberia', 'Newly reported deaths', 7)\n",
    "daily_average_deaths_Liberia_August = avgOverMonth('Liberia', 'Newly reported deaths', 8)\n",
    "daily_average_deaths_Liberia_September = avgOverMonth('Liberia', 'Newly reported deaths', 9)\n",
    "daily_average_deaths_Liberia_October = avgOverMonth('Liberia', 'Newly reported deaths', 10)\n",
    "daily_average_deaths_Liberia_November = avgOverMonth('Liberia', 'Newly reported deaths', 11)\n",
    "daily_average_deaths_Liberia_December = avgOverMonth('Liberia', 'Newly reported deaths', 12)\n",
    "\n",
    "daily_average_newCases_Liberia_June = avgOverMonth('Liberia', ['New Case/s (Suspected)', 'New Case/s (Probable)', 'New case/s (confirmed)'], 6)\n",
    "daily_average_newCases_Liberia_July = avgOverMonth('Liberia', ['New Case/s (Suspected)', 'New Case/s (Probable)', 'New case/s (confirmed)'], 7)\n",
    "daily_average_newCases_Liberia_August = avgOverMonth('Liberia', ['New Case/s (Suspected)', 'New Case/s (Probable)', 'New case/s (confirmed)'], 8)\n",
    "daily_average_newCases_Liberia_September = avgOverMonth('Liberia', ['New Case/s (Suspected)', 'New Case/s (Probable)', 'New case/s (confirmed)'], 9)\n",
    "daily_average_newCases_Liberia_October = avgOverMonth('Liberia', ['New Case/s (Suspected)', 'New Case/s (Probable)', 'New case/s (confirmed)'], 10)\n",
    "daily_average_newCases_Liberia_November = avgOverMonth('Liberia', ['New Case/s (Suspected)', 'New Case/s (Probable)', 'New case/s (confirmed)'], 11)\n",
    "daily_average_newCases_Liberia_December = avgOverMonth('Liberia', ['New Case/s (Suspected)', 'New Case/s (Probable)', 'New case/s (confirmed)'], 12)\n",
    "\n",
    "\n",
    "daily_average_deaths_SL_August = computeAvgNewDeaths('SierraLeone', ['death_suspected', 'death_probable', 'death_confirmed'], 8)\n",
    "daily_average_deaths_SL_September = computeAvgNewDeaths('SierraLeone', ['death_suspected', 'death_probable', 'death_confirmed'], 9)\n",
    "daily_average_deaths_SL_October = computeAvgNewDeaths('SierraLeone', ['death_suspected', 'death_probable', 'death_confirmed'], 10)\n",
    "daily_average_deaths_SL_November = computeAvgNewDeaths('SierraLeone', ['death_suspected', 'death_probable', 'death_confirmed'], 11)\n",
    "daily_average_deaths_SL_December = computeAvgNewDeaths('SierraLeone', ['death_suspected', 'death_probable', 'death_confirmed'], 12)\n",
    "\n",
    "daily_average_newCases_SL_August = avgOverMonth('SierraLeone', ['new_suspected', 'new_probable', 'new_confirmed'], 8)\n",
    "daily_average_newCases_SL_September = avgOverMonth('SierraLeone', ['new_suspected', 'new_probable', 'new_confirmed'], 9)\n",
    "daily_average_newCases_SL_October = avgOverMonth('SierraLeone', ['new_suspected', 'new_probable', 'new_confirmed'], 10)\n",
    "daily_average_newCases_SL_November = avgOverMonth('SierraLeone', ['new_suspected', 'new_probable', 'new_confirmed'], 11)\n",
    "daily_average_newCases_SL_December = avgOverMonth('SierraLeone', ['new_suspected', 'new_probable', 'new_confirmed'], 12)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Country</th>\n",
       "      <th>Description</th>\n",
       "      <th>Month</th>\n",
       "      <th>Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Guinea</td>\n",
       "      <td>Average per day of deaths</td>\n",
       "      <td>August</td>\n",
       "      <td>2.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Guinea</td>\n",
       "      <td>Average per day of deaths</td>\n",
       "      <td>September</td>\n",
       "      <td>3.562500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Guinea</td>\n",
       "      <td>Average per day of deaths</td>\n",
       "      <td>October</td>\n",
       "      <td>15.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Guinea</td>\n",
       "      <td>Average per day of new cases</td>\n",
       "      <td>August</td>\n",
       "      <td>25.800000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Guinea</td>\n",
       "      <td>Average per day of new cases</td>\n",
       "      <td>September</td>\n",
       "      <td>19.625000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Guinea</td>\n",
       "      <td>Average per day of new cases</td>\n",
       "      <td>October</td>\n",
       "      <td>34.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>Average per day of deaths</td>\n",
       "      <td>June</td>\n",
       "      <td>2.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>Average per day of deaths</td>\n",
       "      <td>July</td>\n",
       "      <td>4.272727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>Average per day of deaths</td>\n",
       "      <td>August</td>\n",
       "      <td>23.222222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>Average per day of deaths</td>\n",
       "      <td>September</td>\n",
       "      <td>36.041667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>Average per day of deaths</td>\n",
       "      <td>October</td>\n",
       "      <td>28.040000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>Average per day of deaths</td>\n",
       "      <td>November</td>\n",
       "      <td>13.466667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>Average per day of deaths</td>\n",
       "      <td>December</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>Average per day of new cases</td>\n",
       "      <td>June</td>\n",
       "      <td>5.714286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>Average per day of new cases</td>\n",
       "      <td>July</td>\n",
       "      <td>8.545455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>Average per day of new cases</td>\n",
       "      <td>August</td>\n",
       "      <td>37.222222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>Average per day of new cases</td>\n",
       "      <td>September</td>\n",
       "      <td>63.833333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>Average per day of new cases</td>\n",
       "      <td>October</td>\n",
       "      <td>45.560000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>Average per day of new cases</td>\n",
       "      <td>November</td>\n",
       "      <td>26.466667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>Average per day of new cases</td>\n",
       "      <td>December</td>\n",
       "      <td>5178.555556</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Sierra Leone</td>\n",
       "      <td>Average per day of deaths</td>\n",
       "      <td>August</td>\n",
       "      <td>6.947368</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Sierra Leone</td>\n",
       "      <td>Average per day of deaths</td>\n",
       "      <td>September</td>\n",
       "      <td>5.310345</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Sierra Leone</td>\n",
       "      <td>Average per day of deaths</td>\n",
       "      <td>October</td>\n",
       "      <td>28.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Sierra Leone</td>\n",
       "      <td>Average per day of deaths</td>\n",
       "      <td>November</td>\n",
       "      <td>15.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Sierra Leone</td>\n",
       "      <td>Average per day of deaths</td>\n",
       "      <td>December</td>\n",
       "      <td>31.800000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Sierra Leone</td>\n",
       "      <td>Average per day of new cases</td>\n",
       "      <td>August</td>\n",
       "      <td>25.150000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Sierra Leone</td>\n",
       "      <td>Average per day of new cases</td>\n",
       "      <td>September</td>\n",
       "      <td>40.689655</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Sierra Leone</td>\n",
       "      <td>Average per day of new cases</td>\n",
       "      <td>October</td>\n",
       "      <td>70.928571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Sierra Leone</td>\n",
       "      <td>Average per day of new cases</td>\n",
       "      <td>November</td>\n",
       "      <td>75.238095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Sierra Leone</td>\n",
       "      <td>Average per day of new cases</td>\n",
       "      <td>December</td>\n",
       "      <td>41.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Country                   Description      Month        Value\n",
       "0         Guinea     Average per day of deaths     August     2.000000\n",
       "1         Guinea     Average per day of deaths  September     3.562500\n",
       "2         Guinea     Average per day of deaths    October    15.000000\n",
       "3         Guinea  Average per day of new cases     August    25.800000\n",
       "4         Guinea  Average per day of new cases  September    19.625000\n",
       "5         Guinea  Average per day of new cases    October    34.000000\n",
       "6        Liberia     Average per day of deaths       June     2.000000\n",
       "7        Liberia     Average per day of deaths       July     4.272727\n",
       "8        Liberia     Average per day of deaths     August    23.222222\n",
       "9        Liberia     Average per day of deaths  September    36.041667\n",
       "10       Liberia     Average per day of deaths    October    28.040000\n",
       "11       Liberia     Average per day of deaths   November    13.466667\n",
       "12       Liberia     Average per day of deaths   December     0.000000\n",
       "13       Liberia  Average per day of new cases       June     5.714286\n",
       "14       Liberia  Average per day of new cases       July     8.545455\n",
       "15       Liberia  Average per day of new cases     August    37.222222\n",
       "16       Liberia  Average per day of new cases  September    63.833333\n",
       "17       Liberia  Average per day of new cases    October    45.560000\n",
       "18       Liberia  Average per day of new cases   November    26.466667\n",
       "19       Liberia  Average per day of new cases   December  5178.555556\n",
       "20  Sierra Leone     Average per day of deaths     August     6.947368\n",
       "21  Sierra Leone     Average per day of deaths  September     5.310345\n",
       "22  Sierra Leone     Average per day of deaths    October    28.500000\n",
       "23  Sierra Leone     Average per day of deaths   November    15.000000\n",
       "24  Sierra Leone     Average per day of deaths   December    31.800000\n",
       "25  Sierra Leone  Average per day of new cases     August    25.150000\n",
       "26  Sierra Leone  Average per day of new cases  September    40.689655\n",
       "27  Sierra Leone  Average per day of new cases    October    70.928571\n",
       "28  Sierra Leone  Average per day of new cases   November    75.238095\n",
       "29  Sierra Leone  Average per day of new cases   December    41.000000"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g='Guinea'\n",
    "l='Liberia'\n",
    "sl='Sierra Leone'\n",
    "ad='Average per day of deaths'\n",
    "an='Average per day of new cases'\n",
    "j='June'\n",
    "J='July'\n",
    "a='August'\n",
    "s='September'\n",
    "o='October'\n",
    "n='November'\n",
    "d='December'\n",
    "dfG=pd.DataFrame({'Country':     [g,g,g,g,g,g],\n",
    "                  'Description': [ad,ad,ad,an,an,an],\n",
    "                  'Month':       [a,s,o,a,s,o],\n",
    "                  'Value':       [daily_average_deaths_Guinea_August, daily_average_deaths_Guinea_September, daily_average_deaths_Guinea_October, daily_average_newCases_Guinea_August, daily_average_newCases_Guinea_September, daily_average_newCases_Guinea_October]})\n",
    "dfL=pd.DataFrame({'Country':     [l,l,l,l,l,l,l,l,l,l,l,l,l,l],\n",
    "                  'Description': [ad,ad,ad,ad,ad,ad,ad,an,an,an,an,an,an,an],\n",
    "                  'Month':       [j,J,a,s,o,n,d,j,J,a,s,o,n,d],\n",
    "                  'Value':       [daily_average_deaths_Liberia_June, daily_average_deaths_Liberia_July, daily_average_deaths_Liberia_August, daily_average_deaths_Liberia_September, daily_average_deaths_Liberia_October, daily_average_deaths_Liberia_November, daily_average_deaths_Liberia_December, daily_average_newCases_Liberia_June, daily_average_newCases_Liberia_July, daily_average_newCases_Liberia_August, daily_average_newCases_Liberia_September, daily_average_newCases_Liberia_October, daily_average_newCases_Liberia_November, daily_average_newCases_Liberia_December]})\n",
    "dfSL=pd.DataFrame({'Country':     [sl,sl,sl,sl,sl,sl,sl,sl,sl,sl],\n",
    "                   'Description': [ad,ad,ad,ad,ad,an,an,an,an,an],\n",
    "                   'Month':       [a,s,o,n,d,a,s,o,n,d],\n",
    "                   'Value':       [daily_average_deaths_SL_August, daily_average_deaths_SL_September, daily_average_deaths_SL_October, daily_average_deaths_SL_November, daily_average_deaths_SL_December, daily_average_newCases_SL_August, daily_average_newCases_SL_September, daily_average_newCases_SL_October, daily_average_newCases_SL_November, daily_average_newCases_SL_December]})\n",
    "DF1=pd.merge(dfG,dfL, how='outer')\n",
    "DF=pd.merge(DF1,dfSL, how='outer')\n",
    "DF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 2. RNA Sequences\n",
    "\n",
    "In the `DATA_FOLDER/microbiome` subdirectory, there are 9 spreadsheets of microbiome data that was acquired from high-throughput RNA sequencing procedures, along with a 10<sup>th</sup> file that describes the content of each. \n",
    "\n",
    "Use pandas to import the first 9 spreadsheets into a single `DataFrame`.\n",
    "Then, add the metadata information from the 10<sup>th</sup> spreadsheet as columns in the combined `DataFrame`.\n",
    "Make sure that the final `DataFrame` has a unique index and all the `NaN` values have been replaced by the tag `unknown`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>NAME</th>\n",
       "      <th>VALUE</th>\n",
       "      <th>BARCODE</th>\n",
       "      <th>GROUP</th>\n",
       "      <th>SAMPLE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"30\" valign=\"top\">1</th>\n",
       "      <th>0</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...</td>\n",
       "      <td>7</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...</td>\n",
       "      <td>2</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Sulfoloba...</td>\n",
       "      <td>3</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Thermopro...</td>\n",
       "      <td>3</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...</td>\n",
       "      <td>7</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Archaea \"Euryarchaeota\" Archaeoglobi Archaeogl...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Archaea \"Euryarchaeota\" Archaeoglobi Archaeogl...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Archaea \"Euryarchaeota\" Halobacteria Halobacte...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Archaea \"Euryarchaeota\" Halobacteria Halobacte...</td>\n",
       "      <td>4</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Archaea \"Euryarchaeota\" Halobacteria Halobacte...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Archaea \"Euryarchaeota\" Halobacteria Halobacte...</td>\n",
       "      <td>2</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Archaea \"Euryarchaeota\" Halobacteria Halobacte...</td>\n",
       "      <td>4</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Archaea \"Euryarchaeota\" Methanococci Methanoco...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Archaea \"Euryarchaeota\" Methanopyri Methanopyr...</td>\n",
       "      <td>12</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Archaea \"Euryarchaeota\" Thermoplasmata Thermop...</td>\n",
       "      <td>2</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>2</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>2</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>53</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>2</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>3</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>99</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>17</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>2</td>\n",
       "      <td>MID1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"30\" valign=\"top\">9</th>\n",
       "      <th>251</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>10</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>252</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>253</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>254</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>12</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>4</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>257</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>805</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>258</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>862</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>12</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>3534</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>4</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>262</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>263</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>25</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>264</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>2</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>265</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>3</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>266</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>2</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>3</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>269</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>2</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271</th>\n",
       "      <td>Bacteria \"Synergistetes\" Synergistia Synergist...</td>\n",
       "      <td>3</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>272</th>\n",
       "      <td>Bacteria \"Tenericutes\" Mollicutes Anaeroplasma...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273</th>\n",
       "      <td>Bacteria \"Tenericutes\" Mollicutes Haloplasmata...</td>\n",
       "      <td>11</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>274</th>\n",
       "      <td>Bacteria \"Thermodesulfobacteria\" Thermodesulfo...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275</th>\n",
       "      <td>Bacteria \"Thermodesulfobacteria\" Thermodesulfo...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>276</th>\n",
       "      <td>Bacteria \"Thermotogae\" Thermotogae Thermotogal...</td>\n",
       "      <td>43</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>277</th>\n",
       "      <td>Bacteria \"Thermotogae\" Thermotogae Thermotogal...</td>\n",
       "      <td>16</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>278</th>\n",
       "      <td>Bacteria \"Verrucomicrobia\" Opitutae Opitutales...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>279</th>\n",
       "      <td>Bacteria Cyanobacteria Cyanobacteria  Chloropl...</td>\n",
       "      <td>1</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>280</th>\n",
       "      <td>Bacteria Cyanobacteria Cyanobacteria  Chloropl...</td>\n",
       "      <td>10</td>\n",
       "      <td>MID9</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2396 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    NAME  VALUE BARCODE  \\\n",
       "1 0    Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...      7    MID1   \n",
       "  1    Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...      2    MID1   \n",
       "  2    Archaea \"Crenarchaeota\" Thermoprotei Sulfoloba...      3    MID1   \n",
       "  3    Archaea \"Crenarchaeota\" Thermoprotei Thermopro...      3    MID1   \n",
       "  4    Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...      7    MID1   \n",
       "  5    Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...      1    MID1   \n",
       "  6    Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...      1    MID1   \n",
       "  7    Archaea \"Euryarchaeota\" Archaeoglobi Archaeogl...      1    MID1   \n",
       "  8    Archaea \"Euryarchaeota\" Archaeoglobi Archaeogl...      1    MID1   \n",
       "  9    Archaea \"Euryarchaeota\" Halobacteria Halobacte...      1    MID1   \n",
       "  10   Archaea \"Euryarchaeota\" Halobacteria Halobacte...      4    MID1   \n",
       "  11   Archaea \"Euryarchaeota\" Halobacteria Halobacte...      1    MID1   \n",
       "  12   Archaea \"Euryarchaeota\" Halobacteria Halobacte...      2    MID1   \n",
       "  13   Archaea \"Euryarchaeota\" Halobacteria Halobacte...      4    MID1   \n",
       "  14   Archaea \"Euryarchaeota\" Methanococci Methanoco...      1    MID1   \n",
       "  15   Archaea \"Euryarchaeota\" Methanopyri Methanopyr...     12    MID1   \n",
       "  16   Archaea \"Euryarchaeota\" Thermoplasmata Thermop...      2    MID1   \n",
       "  17   Bacteria \"Actinobacteria\" Actinobacteria Actin...      2    MID1   \n",
       "  18   Bacteria \"Actinobacteria\" Actinobacteria Actin...      1    MID1   \n",
       "  19   Bacteria \"Actinobacteria\" Actinobacteria Actin...      1    MID1   \n",
       "  20   Bacteria \"Actinobacteria\" Actinobacteria Actin...      1    MID1   \n",
       "  21   Bacteria \"Actinobacteria\" Actinobacteria Actin...      2    MID1   \n",
       "  22   Bacteria \"Actinobacteria\" Actinobacteria Actin...     53    MID1   \n",
       "  23   Bacteria \"Actinobacteria\" Actinobacteria Actin...      1    MID1   \n",
       "  24   Bacteria \"Actinobacteria\" Actinobacteria Actin...      2    MID1   \n",
       "  25   Bacteria \"Actinobacteria\" Actinobacteria Actin...      3    MID1   \n",
       "  26   Bacteria \"Actinobacteria\" Actinobacteria Actin...      1    MID1   \n",
       "  27   Bacteria \"Actinobacteria\" Actinobacteria Actin...     99    MID1   \n",
       "  28   Bacteria \"Actinobacteria\" Actinobacteria Actin...     17    MID1   \n",
       "  29   Bacteria \"Actinobacteria\" Actinobacteria Actin...      2    MID1   \n",
       "...                                                  ...    ...     ...   \n",
       "9 251  Bacteria \"Proteobacteria\" Gammaproteobacteria ...     10    MID9   \n",
       "  252  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      1    MID9   \n",
       "  253  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      1    MID9   \n",
       "  254  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      1    MID9   \n",
       "  255  Bacteria \"Proteobacteria\" Gammaproteobacteria ...     12    MID9   \n",
       "  256  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      4    MID9   \n",
       "  257  Bacteria \"Proteobacteria\" Gammaproteobacteria ...    805    MID9   \n",
       "  258  Bacteria \"Proteobacteria\" Gammaproteobacteria ...    862    MID9   \n",
       "  259  Bacteria \"Proteobacteria\" Gammaproteobacteria ...     12    MID9   \n",
       "  260  Bacteria \"Proteobacteria\" Gammaproteobacteria ...   3534    MID9   \n",
       "  261  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      4    MID9   \n",
       "  262  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      1    MID9   \n",
       "  263  Bacteria \"Proteobacteria\" Gammaproteobacteria ...     25    MID9   \n",
       "  264  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      2    MID9   \n",
       "  265  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      3    MID9   \n",
       "  266  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      2    MID9   \n",
       "  267  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      1    MID9   \n",
       "  268  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      3    MID9   \n",
       "  269  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      2    MID9   \n",
       "  270  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      1    MID9   \n",
       "  271  Bacteria \"Synergistetes\" Synergistia Synergist...      3    MID9   \n",
       "  272  Bacteria \"Tenericutes\" Mollicutes Anaeroplasma...      1    MID9   \n",
       "  273  Bacteria \"Tenericutes\" Mollicutes Haloplasmata...     11    MID9   \n",
       "  274  Bacteria \"Thermodesulfobacteria\" Thermodesulfo...      1    MID9   \n",
       "  275  Bacteria \"Thermodesulfobacteria\" Thermodesulfo...      1    MID9   \n",
       "  276  Bacteria \"Thermotogae\" Thermotogae Thermotogal...     43    MID9   \n",
       "  277  Bacteria \"Thermotogae\" Thermotogae Thermotogal...     16    MID9   \n",
       "  278  Bacteria \"Verrucomicrobia\" Opitutae Opitutales...      1    MID9   \n",
       "  279  Bacteria Cyanobacteria Cyanobacteria  Chloropl...      1    MID9   \n",
       "  280  Bacteria Cyanobacteria Cyanobacteria  Chloropl...     10    MID9   \n",
       "\n",
       "                    GROUP   SAMPLE  \n",
       "1 0    EXTRACTION CONTROL  unknown  \n",
       "  1    EXTRACTION CONTROL  unknown  \n",
       "  2    EXTRACTION CONTROL  unknown  \n",
       "  3    EXTRACTION CONTROL  unknown  \n",
       "  4    EXTRACTION CONTROL  unknown  \n",
       "  5    EXTRACTION CONTROL  unknown  \n",
       "  6    EXTRACTION CONTROL  unknown  \n",
       "  7    EXTRACTION CONTROL  unknown  \n",
       "  8    EXTRACTION CONTROL  unknown  \n",
       "  9    EXTRACTION CONTROL  unknown  \n",
       "  10   EXTRACTION CONTROL  unknown  \n",
       "  11   EXTRACTION CONTROL  unknown  \n",
       "  12   EXTRACTION CONTROL  unknown  \n",
       "  13   EXTRACTION CONTROL  unknown  \n",
       "  14   EXTRACTION CONTROL  unknown  \n",
       "  15   EXTRACTION CONTROL  unknown  \n",
       "  16   EXTRACTION CONTROL  unknown  \n",
       "  17   EXTRACTION CONTROL  unknown  \n",
       "  18   EXTRACTION CONTROL  unknown  \n",
       "  19   EXTRACTION CONTROL  unknown  \n",
       "  20   EXTRACTION CONTROL  unknown  \n",
       "  21   EXTRACTION CONTROL  unknown  \n",
       "  22   EXTRACTION CONTROL  unknown  \n",
       "  23   EXTRACTION CONTROL  unknown  \n",
       "  24   EXTRACTION CONTROL  unknown  \n",
       "  25   EXTRACTION CONTROL  unknown  \n",
       "  26   EXTRACTION CONTROL  unknown  \n",
       "  27   EXTRACTION CONTROL  unknown  \n",
       "  28   EXTRACTION CONTROL  unknown  \n",
       "  29   EXTRACTION CONTROL  unknown  \n",
       "...                   ...      ...  \n",
       "9 251           Control 2    stool  \n",
       "  252           Control 2    stool  \n",
       "  253           Control 2    stool  \n",
       "  254           Control 2    stool  \n",
       "  255           Control 2    stool  \n",
       "  256           Control 2    stool  \n",
       "  257           Control 2    stool  \n",
       "  258           Control 2    stool  \n",
       "  259           Control 2    stool  \n",
       "  260           Control 2    stool  \n",
       "  261           Control 2    stool  \n",
       "  262           Control 2    stool  \n",
       "  263           Control 2    stool  \n",
       "  264           Control 2    stool  \n",
       "  265           Control 2    stool  \n",
       "  266           Control 2    stool  \n",
       "  267           Control 2    stool  \n",
       "  268           Control 2    stool  \n",
       "  269           Control 2    stool  \n",
       "  270           Control 2    stool  \n",
       "  271           Control 2    stool  \n",
       "  272           Control 2    stool  \n",
       "  273           Control 2    stool  \n",
       "  274           Control 2    stool  \n",
       "  275           Control 2    stool  \n",
       "  276           Control 2    stool  \n",
       "  277           Control 2    stool  \n",
       "  278           Control 2    stool  \n",
       "  279           Control 2    stool  \n",
       "  280           Control 2    stool  \n",
       "\n",
       "[2396 rows x 5 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Can't use glob here because it does not load the sheets in a precise order, so we do it manually\n",
    "paths = [PATH_MICROBIOME + \"/MID1.xls\",\n",
    "         PATH_MICROBIOME + \"/MID2.xls\",\n",
    "         PATH_MICROBIOME + \"/MID3.xls\",\n",
    "         PATH_MICROBIOME + \"/MID4.xls\",\n",
    "         PATH_MICROBIOME + \"/MID5.xls\",\n",
    "         PATH_MICROBIOME + \"/MID6.xls\",\n",
    "         PATH_MICROBIOME + \"/MID7.xls\",\n",
    "         PATH_MICROBIOME + \"/MID8.xls\",\n",
    "         PATH_MICROBIOME + \"/MID9.xls\"]\n",
    "\n",
    "data_array = [pd.read_excel(path, sheetname='Sheet 1', header=None, na_values=na_val, fill_value='unknown') for path in paths]\n",
    "df = pd.concat(data_array, keys=[i+1 for i in range(len(data_array))])\n",
    "df.rename(columns={0:'NAME', 1:'VALUE'}, inplace=True)\n",
    "\n",
    "metadata = pd.read_excel(PATH_MICROBIOME + \"/metadata.xls\", na_values=na_val, fill_value='unknown')\n",
    "for item in metadata.items():\n",
    "    df[item[0]] = 'unknown'\n",
    "    for i in range(1,10):\n",
    "        df.loc[i][item[0]] = item[1][i-1]\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 3. Class War in Titanic\n",
    "\n",
    "Use pandas to import the data file `Data/titanic.xls`. It contains data on all the passengers that travelled on the Titanic."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from IPython.core.display import HTML\n",
    "HTML(filename=DATA_FOLDER+'/titanic.html')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For each of the following questions state clearly your assumptions and discuss your findings:\n",
    "1. Describe the *type* and the *value range* of each attribute. Indicate and transform the attributes that can be `Categorical`. \n",
    "2. Plot histograms for the *travel class*, *embarkation port*, *sex* and *age* attributes. For the latter one, use *discrete decade intervals*. \n",
    "3. Calculate the proportion of passengers by *cabin floor*. Present your results in a *pie chart*.\n",
    "4. For each *travel class*, calculate the proportion of the passengers that survived. Present your results in *pie charts*.\n",
    "5. Calculate the proportion of the passengers that survived by *travel class* and *sex*. Present your results in *a single histogram*.\n",
    "6. Create 2 equally populated *age categories* and calculate survival proportions by *age category*, *travel class* and *sex*. Present your results in a `DataFrame` with unique index."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Write your answer here"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
