{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Table of Contents\n",
    " <p><div class=\"lev1\"><a href=\"#Task-1.-Compiling-Ebola-Data\"><span class=\"toc-item-num\">Task 1.&nbsp;&nbsp;</span>Compiling Ebola Data</a></div>\n",
    " <div class=\"lev1\"><a href=\"#Task-2.-RNA-Sequences\"><span class=\"toc-item-num\">Task 2.&nbsp;&nbsp;</span>RNA Sequences</a></div>\n",
    " <div class=\"lev1\"><a href=\"#Task-3.-Class-War-in-Titanic\"><span class=\"toc-item-num\">Task 3.&nbsp;&nbsp;</span>Class War in Titanic</a></div></p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "DATA_FOLDER = 'Data' # Use the data folder provided in Tutorial 02 - Intro to Pandas."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 1. Compiling Ebola Data\n",
    "\n",
    "The `DATA_FOLDER/ebola` folder contains summarized reports of Ebola cases from three countries (Guinea, Liberia and Sierra Leone) during the recent outbreak of the disease in West Africa. For each country, there are daily reports that contain various information about the outbreak in several cities in each country.\n",
    "\n",
    "Use pandas to import these data files into a single `Dataframe`.\n",
    "Using this `DataFrame`, calculate for *each country*, the *daily average per month* of *new cases* and *deaths*.\n",
    "Make sure you handle all the different expressions for *new cases* and *deaths* that are used in the reports."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Data/ebola/guinea_data\\\\2014-08-04.csv', 'Data/ebola/guinea_data\\\\2014-08-26.csv', 'Data/ebola/guinea_data\\\\2014-08-27.csv', 'Data/ebola/guinea_data\\\\2014-08-30.csv', 'Data/ebola/guinea_data\\\\2014-08-31.csv', 'Data/ebola/guinea_data\\\\2014-09-02.csv', 'Data/ebola/guinea_data\\\\2014-09-04.csv', 'Data/ebola/guinea_data\\\\2014-09-07.csv', 'Data/ebola/guinea_data\\\\2014-09-08.csv', 'Data/ebola/guinea_data\\\\2014-09-09.csv', 'Data/ebola/guinea_data\\\\2014-09-11.csv', 'Data/ebola/guinea_data\\\\2014-09-14.csv', 'Data/ebola/guinea_data\\\\2014-09-16.csv', 'Data/ebola/guinea_data\\\\2014-09-17.csv', 'Data/ebola/guinea_data\\\\2014-09-19.csv', 'Data/ebola/guinea_data\\\\2014-09-21.csv', 'Data/ebola/guinea_data\\\\2014-09-22.csv', 'Data/ebola/guinea_data\\\\2014-09-23.csv', 'Data/ebola/guinea_data\\\\2014-09-24.csv', 'Data/ebola/guinea_data\\\\2014-09-26.csv', 'Data/ebola/guinea_data\\\\2014-09-30.csv', 'Data/ebola/guinea_data\\\\2014-10-01.csv']\n",
      "['Data/ebola/sl_data\\\\2014-08-12-v77.csv', 'Data/ebola/sl_data\\\\2014-08-13-v78.csv', 'Data/ebola/sl_data\\\\2014-08-14-v79.csv', 'Data/ebola/sl_data\\\\2014-08-15-v80.csv', 'Data/ebola/sl_data\\\\2014-08-16-v81.csv', 'Data/ebola/sl_data\\\\2014-08-17-v82.csv', 'Data/ebola/sl_data\\\\2014-08-18-v83.csv', 'Data/ebola/sl_data\\\\2014-08-19-v84.csv', 'Data/ebola/sl_data\\\\2014-08-20-v85.csv', 'Data/ebola/sl_data\\\\2014-08-21-v86.csv', 'Data/ebola/sl_data\\\\2014-08-22-v87.csv', 'Data/ebola/sl_data\\\\2014-08-23-v88.csv', 'Data/ebola/sl_data\\\\2014-08-24-v89.csv', 'Data/ebola/sl_data\\\\2014-08-25-v90.csv', 'Data/ebola/sl_data\\\\2014-08-26-v91.csv', 'Data/ebola/sl_data\\\\2014-08-27-v92.csv', 'Data/ebola/sl_data\\\\2014-08-28-v93.csv', 'Data/ebola/sl_data\\\\2014-08-29-v94.csv', 'Data/ebola/sl_data\\\\2014-08-30-v95.csv', 'Data/ebola/sl_data\\\\2014-08-31-v96.csv', 'Data/ebola/sl_data\\\\2014-09-01-v97.csv', 'Data/ebola/sl_data\\\\2014-09-02-v98.csv', 'Data/ebola/sl_data\\\\2014-09-03-v99.csv', 'Data/ebola/sl_data\\\\2014-09-05-v101.csv', 'Data/ebola/sl_data\\\\2014-09-06-v102.csv', 'Data/ebola/sl_data\\\\2014-09-07-v103.csv', 'Data/ebola/sl_data\\\\2014-09-08-v104.csv', 'Data/ebola/sl_data\\\\2014-09-09-v105.csv', 'Data/ebola/sl_data\\\\2014-09-10-v106.csv', 'Data/ebola/sl_data\\\\2014-09-11-v107.csv', 'Data/ebola/sl_data\\\\2014-09-12-v108.csv', 'Data/ebola/sl_data\\\\2014-09-13-v109.csv', 'Data/ebola/sl_data\\\\2014-09-14-v110.csv', 'Data/ebola/sl_data\\\\2014-09-15-v111.csv', 'Data/ebola/sl_data\\\\2014-09-16-v112.csv', 'Data/ebola/sl_data\\\\2014-09-17-v113.csv', 'Data/ebola/sl_data\\\\2014-09-18-v114.csv', 'Data/ebola/sl_data\\\\2014-09-19-v115.csv', 'Data/ebola/sl_data\\\\2014-09-20-v116.csv', 'Data/ebola/sl_data\\\\2014-09-21-v117.csv', 'Data/ebola/sl_data\\\\2014-09-22-v118.csv', 'Data/ebola/sl_data\\\\2014-09-23-v119.csv', 'Data/ebola/sl_data\\\\2014-09-24-v120.csv', 'Data/ebola/sl_data\\\\2014-09-25-v121.csv', 'Data/ebola/sl_data\\\\2014-09-26-v122.csv', 'Data/ebola/sl_data\\\\2014-09-27-v123.csv', 'Data/ebola/sl_data\\\\2014-09-28-v124.csv', 'Data/ebola/sl_data\\\\2014-09-29-v125.csv', 'Data/ebola/sl_data\\\\2014-09-30-v126.csv', 'Data/ebola/sl_data\\\\2014-10-01-v127.csv', 'Data/ebola/sl_data\\\\2014-10-03-v129.csv', 'Data/ebola/sl_data\\\\2014-10-04-v130.csv', 'Data/ebola/sl_data\\\\2014-10-05-v131.csv', 'Data/ebola/sl_data\\\\2014-10-06-v132.csv', 'Data/ebola/sl_data\\\\2014-10-07-v133.csv', 'Data/ebola/sl_data\\\\2014-10-08-v134.csv', 'Data/ebola/sl_data\\\\2014-10-10-v136.csv', 'Data/ebola/sl_data\\\\2014-10-11-v137.csv', 'Data/ebola/sl_data\\\\2014-10-12-v138.csv', 'Data/ebola/sl_data\\\\2014-10-13-v139.csv', 'Data/ebola/sl_data\\\\2014-10-14-v140.csv', 'Data/ebola/sl_data\\\\2014-10-16-v141.csv', 'Data/ebola/sl_data\\\\2014-10-16-v142.csv', 'Data/ebola/sl_data\\\\2014-10-17-v143.csv', 'Data/ebola/sl_data\\\\2014-10-18-v144.csv', 'Data/ebola/sl_data\\\\2014-10-19-v145.csv', 'Data/ebola/sl_data\\\\2014-10-20-v146.csv', 'Data/ebola/sl_data\\\\2014-10-21-v147.csv', 'Data/ebola/sl_data\\\\2014-10-22-v148.csv', 'Data/ebola/sl_data\\\\2014-10-23-v149.csv', 'Data/ebola/sl_data\\\\2014-10-24-v150.csv', 'Data/ebola/sl_data\\\\2014-10-25-v151.csv', 'Data/ebola/sl_data\\\\2014-10-26-v152.csv', 'Data/ebola/sl_data\\\\2014-10-27-v153.csv', 'Data/ebola/sl_data\\\\2014-10-28-v154.csv', 'Data/ebola/sl_data\\\\2014-10-30-v156.csv', 'Data/ebola/sl_data\\\\2014-10-31-v157.csv', 'Data/ebola/sl_data\\\\2014-11-01-v158.csv', 'Data/ebola/sl_data\\\\2014-11-02-v159.csv', 'Data/ebola/sl_data\\\\2014-11-06-v163.csv', 'Data/ebola/sl_data\\\\2014-11-07-v164.csv', 'Data/ebola/sl_data\\\\2014-11-08-v165.csv', 'Data/ebola/sl_data\\\\2014-11-10-v167.csv', 'Data/ebola/sl_data\\\\2014-11-12-v169.csv', 'Data/ebola/sl_data\\\\2014-11-13.csv', 'Data/ebola/sl_data\\\\2014-11-14.csv', 'Data/ebola/sl_data\\\\2014-11-15.csv', 'Data/ebola/sl_data\\\\2014-11-16.csv', 'Data/ebola/sl_data\\\\2014-11-17.csv', 'Data/ebola/sl_data\\\\2014-11-18-v175.csv', 'Data/ebola/sl_data\\\\2014-11-19-v176.csv', 'Data/ebola/sl_data\\\\2014-11-20-v177.csv', 'Data/ebola/sl_data\\\\2014-11-21-v178.csv', 'Data/ebola/sl_data\\\\2014-11-22-v179.csv', 'Data/ebola/sl_data\\\\2014-11-23-v180.csv', 'Data/ebola/sl_data\\\\2014-11-24-v181.csv', 'Data/ebola/sl_data\\\\2014-11-28.csv', 'Data/ebola/sl_data\\\\2014-11-29.csv', 'Data/ebola/sl_data\\\\2014-12-01.csv', 'Data/ebola/sl_data\\\\2014-12-04.csv', 'Data/ebola/sl_data\\\\2014-12-05.csv', 'Data/ebola/sl_data\\\\2014-12-06.csv', 'Data/ebola/sl_data\\\\2014-12-13.csv']\n",
      "['Data/ebola/liberia_data\\\\2014-06-16.csv', 'Data/ebola/liberia_data\\\\2014-06-17.csv', 'Data/ebola/liberia_data\\\\2014-06-22.csv', 'Data/ebola/liberia_data\\\\2014-06-24.csv', 'Data/ebola/liberia_data\\\\2014-06-25.csv', 'Data/ebola/liberia_data\\\\2014-06-28.csv', 'Data/ebola/liberia_data\\\\2014-06-29.csv', 'Data/ebola/liberia_data\\\\2014-07-01.csv', 'Data/ebola/liberia_data\\\\2014-07-02.csv', 'Data/ebola/liberia_data\\\\2014-07-03.csv', 'Data/ebola/liberia_data\\\\2014-07-07.csv', 'Data/ebola/liberia_data\\\\2014-07-08.csv', 'Data/ebola/liberia_data\\\\2014-07-10.csv', 'Data/ebola/liberia_data\\\\2014-07-13.csv', 'Data/ebola/liberia_data\\\\2014-07-17.csv', 'Data/ebola/liberia_data\\\\2014-07-20.csv', 'Data/ebola/liberia_data\\\\2014-07-24.csv', 'Data/ebola/liberia_data\\\\2014-07-26.csv', 'Data/ebola/liberia_data\\\\2014-08-02.csv', 'Data/ebola/liberia_data\\\\2014-08-04.csv', 'Data/ebola/liberia_data\\\\2014-08-12.csv', 'Data/ebola/liberia_data\\\\2014-08-15.csv', 'Data/ebola/liberia_data\\\\2014-08-17.csv', 'Data/ebola/liberia_data\\\\2014-08-18.csv', 'Data/ebola/liberia_data\\\\2014-08-20.csv', 'Data/ebola/liberia_data\\\\2014-08-25.csv', 'Data/ebola/liberia_data\\\\2014-08-28.csv', 'Data/ebola/liberia_data\\\\2014-09-01.csv', 'Data/ebola/liberia_data\\\\2014-09-02.csv', 'Data/ebola/liberia_data\\\\2014-09-03.csv', 'Data/ebola/liberia_data\\\\2014-09-04.csv', 'Data/ebola/liberia_data\\\\2014-09-05.csv', 'Data/ebola/liberia_data\\\\2014-09-06.csv', 'Data/ebola/liberia_data\\\\2014-09-07.csv', 'Data/ebola/liberia_data\\\\2014-09-08.csv', 'Data/ebola/liberia_data\\\\2014-09-10.csv', 'Data/ebola/liberia_data\\\\2014-09-11.csv', 'Data/ebola/liberia_data\\\\2014-09-12.csv', 'Data/ebola/liberia_data\\\\2014-09-13.csv', 'Data/ebola/liberia_data\\\\2014-09-14.csv', 'Data/ebola/liberia_data\\\\2014-09-15.csv', 'Data/ebola/liberia_data\\\\2014-09-16.csv', 'Data/ebola/liberia_data\\\\2014-09-17-v125.csv', 'Data/ebola/liberia_data\\\\2014-09-20-v128.csv', 'Data/ebola/liberia_data\\\\2014-09-21-v129.csv', 'Data/ebola/liberia_data\\\\2014-09-23-v131.csv', 'Data/ebola/liberia_data\\\\2014-09-25-v133.csv', 'Data/ebola/liberia_data\\\\2014-09-26-v134.csv', 'Data/ebola/liberia_data\\\\2014-09-27-v135.csv', 'Data/ebola/liberia_data\\\\2014-09-28-v136.csv', 'Data/ebola/liberia_data\\\\2014-09-30-v138.csv', 'Data/ebola/liberia_data\\\\2014-10-01-v139.csv', 'Data/ebola/liberia_data\\\\2014-10-03-v141.csv', 'Data/ebola/liberia_data\\\\2014-10-04-v142.csv', 'Data/ebola/liberia_data\\\\2014-10-05-v143.csv', 'Data/ebola/liberia_data\\\\2014-10-07-v145.csv', 'Data/ebola/liberia_data\\\\2014-10-08-v146.csv', 'Data/ebola/liberia_data\\\\2014-10-09-v147.csv', 'Data/ebola/liberia_data\\\\2014-10-10-v148.csv', 'Data/ebola/liberia_data\\\\2014-10-11-v149.csv', 'Data/ebola/liberia_data\\\\2014-10-12-v150.csv', 'Data/ebola/liberia_data\\\\2014-10-13-v151.csv', 'Data/ebola/liberia_data\\\\2014-10-16-v154.csv', 'Data/ebola/liberia_data\\\\2014-10-17-v155.csv', 'Data/ebola/liberia_data\\\\2014-10-18-v156.csv', 'Data/ebola/liberia_data\\\\2014-10-19-v157.csv', 'Data/ebola/liberia_data\\\\2014-10-20-v158.csv', 'Data/ebola/liberia_data\\\\2014-10-21-v159.csv', 'Data/ebola/liberia_data\\\\2014-10-22-v160.csv', 'Data/ebola/liberia_data\\\\2014-10-23-v161.csv', 'Data/ebola/liberia_data\\\\2014-10-24-v162.csv', 'Data/ebola/liberia_data\\\\2014-10-25-v163.csv', 'Data/ebola/liberia_data\\\\2014-10-28-v166.csv', 'Data/ebola/liberia_data\\\\2014-10-29-v167.csv', 'Data/ebola/liberia_data\\\\2014-10-30-v168.csv', 'Data/ebola/liberia_data\\\\2014-10-31-v169.csv', 'Data/ebola/liberia_data\\\\2014-11-02-v171.csv', 'Data/ebola/liberia_data\\\\2014-11-04-v173.csv', 'Data/ebola/liberia_data\\\\2014-11-08-v177.csv', 'Data/ebola/liberia_data\\\\2014-11-14-v183.csv', 'Data/ebola/liberia_data\\\\2014-11-15-v184.csv', 'Data/ebola/liberia_data\\\\2014-11-19-v188.csv', 'Data/ebola/liberia_data\\\\2014-11-20-v189.csv', 'Data/ebola/liberia_data\\\\2014-11-21-v190.csv', 'Data/ebola/liberia_data\\\\2014-11-23-v192.csv', 'Data/ebola/liberia_data\\\\2014-11-24-v193.csv', 'Data/ebola/liberia_data\\\\2014-11-26.csv', 'Data/ebola/liberia_data\\\\2014-11-27.csv', 'Data/ebola/liberia_data\\\\2014-11-28.csv', 'Data/ebola/liberia_data\\\\2014-11-29.csv', 'Data/ebola/liberia_data\\\\2014-11-30.csv', 'Data/ebola/liberia_data\\\\2014-12-01.csv', 'Data/ebola/liberia_data\\\\2014-12-02.csv', 'Data/ebola/liberia_data\\\\2014-12-03.csv', 'Data/ebola/liberia_data\\\\2014-12-04.csv', 'Data/ebola/liberia_data\\\\2014-12-05.csv', 'Data/ebola/liberia_data\\\\2014-12-06.csv', 'Data/ebola/liberia_data\\\\2014-12-07.csv', 'Data/ebola/liberia_data\\\\2014-12-08.csv', 'Data/ebola/liberia_data\\\\2014-12-09.csv']\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1. Guinea\n",
      "Average daily deaths per month\n",
      "Month\n",
      "8.0      3.4000\n",
      "9.0      3.5625\n",
      "10.0    15.0000\n",
      "Name: Totals, dtype: float64 \n",
      "\n",
      "Average daily new cases per month\n",
      "Month\n",
      "8.0     25.800\n",
      "9.0     19.625\n",
      "10.0    34.000\n",
      "Name: Totals, dtype: float64 \n",
      "\n",
      "2. Liberia\n",
      "Average daily deaths per month\n",
      "Month\n",
      "6.0      2.000000\n",
      "7.0      4.272727\n",
      "8.0     23.222222\n",
      "9.0     36.041667\n",
      "10.0    28.040000\n",
      "11.0    13.466667\n",
      "Name: Totals, dtype: float64 \n",
      "\n",
      "Average daily new cases per month\n",
      "Month\n",
      "6.0      5.714286\n",
      "7.0      8.545455\n",
      "8.0     37.222222\n",
      "9.0     63.833333\n",
      "10.0    45.560000\n",
      "11.0    26.466667\n",
      "Name: Totals, dtype: float64 \n",
      "\n",
      "3. Sierra Leone\n",
      "Average daily new cases per month\n",
      "Month\n",
      "8.0     25.150000\n",
      "9.0     40.689655\n",
      "10.0    70.928571\n",
      "11.0    75.238095\n",
      "12.0    41.000000\n",
      "Name: Totals, dtype: float64\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import glob as glob\n",
    "import datetime as dt\n",
    "\n",
    "def read_country(country):\n",
    "    files = glob.glob(DATA_FOLDER + \"/ebola/\" + country + \"/*.csv\")\n",
    "    df = pd.concat((pd.read_csv(file) for file in files))\n",
    "    df['Country'] = country\n",
    "    return df.fillna(0)\n",
    "\n",
    "def read_countries():\n",
    "    df = read_country(\"guinea_data\")\n",
    "    df = df.append(read_country(\"sl_data\"))\n",
    "    df = df.append(read_country(\"liberia_data\"))\n",
    "    return df\n",
    "\n",
    "\n",
    "def normalize(df):\n",
    "    df['Date'].fillna(df['date'], inplace=True)\n",
    "    df['Month'] = df['Date'].map(getMonth)\n",
    "    df['Day'] = df['Date'].map(getDay)\n",
    "    df['Description'].fillna(df['Variable'], inplace=True)\n",
    "    df['Description'].fillna(df['variable'], inplace=True)\n",
    "    df['Totals'].fillna(df['National'], inplace=True)\n",
    "    del df['Date']\n",
    "    del df['date']\n",
    "    del df['Variable']\n",
    "    del df['variable']\n",
    "    del df['National']\n",
    "    \n",
    "def getMonth(date):\n",
    "    if ('/' in date):\n",
    "        try :\n",
    "            d = dt.datetime.strptime(date, \"%m/%d/%Y\")\n",
    "        except ValueError:\n",
    "            d = dt.datetime.strptime(date, \"%m/%d/%y\")\n",
    "    else: d = dt.datetime.strptime(date, \"%Y-%m-%d\")\n",
    "    return d.month\n",
    "\n",
    "def getDay(date):\n",
    "    if ('/' in date):\n",
    "        try :\n",
    "            d = dt.datetime.strptime(date, \"%m/%d/%Y\")\n",
    "        except ValueError:\n",
    "            d = dt.datetime.strptime(date, \"%m/%d/%y\")\n",
    "    else: d = dt.datetime.strptime(date, \"%Y-%m-%d\")\n",
    "    return d.day\n",
    "\n",
    "\n",
    "def compute_means(tots):\n",
    "    tots = tots.astype(float)\n",
    "    return tots.groupby(['Month'])['Totals'].mean()\n",
    "\n",
    "def guinea_death_means(guin):\n",
    "    print(\"Average daily deaths per month\")\n",
    "    guinTots = guin[(guin['Description'] == \"New deaths registered today\")\n",
    "                   | (guin['Description'] == \"New deaths registered\")][['Totals', 'Month']]\n",
    "    return compute_means(guinTots)\n",
    "    \n",
    "def guinea_nc_means(guin):    \n",
    "    print(\"Average daily new cases per month\")\n",
    "    guinTots = guin[guin['Description'] == \"Total new cases registered so far\"][['Totals', 'Month']]\n",
    "    return compute_means(guinTots)\n",
    "\n",
    "def liberia_death_means(lib):\n",
    "    print(\"Average daily deaths per month\")\n",
    "    lib = lib[lib['Month'] < 12]\n",
    "    libTots = lib[lib['Description'] == \"Newly reported deaths\"][['Totals', 'Month']]\n",
    "    return compute_means(libTots)\n",
    "\n",
    "def liberia_nc_means(lib):   \n",
    "    print(\"Average daily new cases per month\")\n",
    "    lib = lib[lib['Month'] < 12]\n",
    "    libTots = lib[(lib['Description'] == \"New Case/s (Suspected)\")\n",
    "                 | (lib['Description'] == \"New Case/s (Probable)\")\n",
    "                 | (lib['Description'] == \"New case/s (confirmed)\")][['Totals', 'Month']]\n",
    "    return compute_means(libTots) * 3\n",
    "    \n",
    "def sl_death_means(sl):\n",
    "    return 0\n",
    "\n",
    "def sl_nc_means(sl):   \n",
    "    print(\"Average daily new cases per month\")\n",
    "    slTots = sl[(sl['Description'] == \"new_suspected\")\n",
    "                 | (sl['Description'] == \"new_probable\")\n",
    "                 | (sl['Description'] == \"new_confirmed\")][['Totals', 'Month']]\n",
    "    return compute_means(slTots) * 3\n",
    "\n",
    "\n",
    "df = read_countries()\n",
    "\n",
    "# check that the sum of the columns is correct\n",
    "# What if the variables are different?\n",
    "normalize(df)\n",
    "print(\"1. Guinea\")\n",
    "guin = df[df['Country'] == \"guinea_data\"]\n",
    "print(guinea_death_means(guin), \"\\n\")\n",
    "print(guinea_nc_means(guin), \"\\n\")\n",
    "print(\"2. Liberia\")\n",
    "lib = df[df['Country'] == \"liberia_data\"]\n",
    "print(liberia_death_means(lib), \"\\n\")\n",
    "print(liberia_nc_means(lib), \"\\n\")\n",
    "print(\"3. Sierra Leone\")\n",
    "sl = df[df['Country'] == \"sl_data\"]\n",
    "#sl_death_means(sl)\n",
    "print(sl_nc_means(sl))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 2. RNA Sequences\n",
    "\n",
    "In the `DATA_FOLDER/microbiome` subdirectory, there are 9 spreadsheets of microbiome data that was acquired from high-throughput RNA sequencing procedures, along with a 10<sup>th</sup> file that describes the content of each. \n",
    "\n",
    "Use pandas to import the first 9 spreadsheets into a single `DataFrame`.\n",
    "Then, add the metadata information from the 10<sup>th</sup> spreadsheet as columns in the combined `DataFrame`.\n",
    "Make sure that the final `DataFrame` has a unique index and all the `NaN` values have been replaced by the tag `unknown`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Michele\\Anaconda3\\lib\\site-packages\\pandas\\core\\indexes\\api.py:77: RuntimeWarning: '>' not supported between instances of 'int' and 'str', sort order is undefined for incomparable objects\n",
      "  result = result.union(other)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BARCODE</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Sulfoloba...</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Thermopro...</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Archaea \"Euryarchaeota\" Archaeoglobi Archaeogl...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Archaea \"Euryarchaeota\" Archaeoglobi Archaeogl...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Archaea \"Euryarchaeota\" Halobacteria Halobacte...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Archaea \"Euryarchaeota\" Halobacteria Halobacte...</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Archaea \"Euryarchaeota\" Halobacteria Halobacte...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Archaea \"Euryarchaeota\" Halobacteria Halobacte...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Archaea \"Euryarchaeota\" Halobacteria Halobacte...</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Archaea \"Euryarchaeota\" Methanococci Methanoco...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Archaea \"Euryarchaeota\" Methanopyri Methanopyr...</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Archaea \"Euryarchaeota\" Thermoplasmata Thermop...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>53.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>99.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>MID1</td>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Firmicutes\" \"Clostridia\" Clostridial...</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Firmicutes\" \"Clostridia\" Clostridial...</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Firmicutes\" \"Clostridia\" Clostridial...</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Firmicutes\" \"Clostridia\" Clostridial...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Firmicutes\" \"Clostridia\" Clostridial...</td>\n",
       "      <td>616.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Firmicutes\" \"Erysipelotrichi\" \"Erysi...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Fusobacteria\" \"Fusobacteria\" \"Fusoba...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Nitrospira\" \"Nitrospira\" \"Nitrospira...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Planctomycetes\" \"Planctomycetacia\" P...</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Proteobacteria\" Alphaproteobacteria ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Proteobacteria\" Betaproteobacteria B...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Proteobacteria\" Betaproteobacteria B...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Proteobacteria\" Betaproteobacteria H...</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Proteobacteria\" Deltaproteobacteria ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Proteobacteria\" Deltaproteobacteria ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Proteobacteria\" Deltaproteobacteria ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Proteobacteria\" Deltaproteobacteria ...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>22.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Thermodesulfobacteria\" Thermodesulfo...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Thermotogae\" Thermotogae Thermotogal...</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria \"Thermotogae\" Thermotogae Thermotogal...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria Cyanobacteria Cyanobacteria  Chloropl...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>MID8</td>\n",
       "      <td>Bacteria Cyanobacteria Cyanobacteria  Chloropl...</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2115 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   BARCODE                                                  0      1\n",
       "0     MID1  Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...    7.0\n",
       "1     MID1  Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...    2.0\n",
       "2     MID1  Archaea \"Crenarchaeota\" Thermoprotei Sulfoloba...    3.0\n",
       "3     MID1  Archaea \"Crenarchaeota\" Thermoprotei Thermopro...    3.0\n",
       "4     MID1  Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...    7.0\n",
       "5     MID1  Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...    1.0\n",
       "6     MID1  Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...    1.0\n",
       "7     MID1  Archaea \"Euryarchaeota\" Archaeoglobi Archaeogl...    1.0\n",
       "8     MID1  Archaea \"Euryarchaeota\" Archaeoglobi Archaeogl...    1.0\n",
       "9     MID1  Archaea \"Euryarchaeota\" Halobacteria Halobacte...    1.0\n",
       "10    MID1  Archaea \"Euryarchaeota\" Halobacteria Halobacte...    4.0\n",
       "11    MID1  Archaea \"Euryarchaeota\" Halobacteria Halobacte...    1.0\n",
       "12    MID1  Archaea \"Euryarchaeota\" Halobacteria Halobacte...    2.0\n",
       "13    MID1  Archaea \"Euryarchaeota\" Halobacteria Halobacte...    4.0\n",
       "14    MID1  Archaea \"Euryarchaeota\" Methanococci Methanoco...    1.0\n",
       "15    MID1  Archaea \"Euryarchaeota\" Methanopyri Methanopyr...   12.0\n",
       "16    MID1  Archaea \"Euryarchaeota\" Thermoplasmata Thermop...    2.0\n",
       "17    MID1  Bacteria \"Actinobacteria\" Actinobacteria Actin...    2.0\n",
       "18    MID1  Bacteria \"Actinobacteria\" Actinobacteria Actin...    1.0\n",
       "19    MID1  Bacteria \"Actinobacteria\" Actinobacteria Actin...    1.0\n",
       "20    MID1  Bacteria \"Actinobacteria\" Actinobacteria Actin...    1.0\n",
       "21    MID1  Bacteria \"Actinobacteria\" Actinobacteria Actin...    2.0\n",
       "22    MID1  Bacteria \"Actinobacteria\" Actinobacteria Actin...   53.0\n",
       "23    MID1  Bacteria \"Actinobacteria\" Actinobacteria Actin...    1.0\n",
       "24    MID1  Bacteria \"Actinobacteria\" Actinobacteria Actin...    2.0\n",
       "25    MID1  Bacteria \"Actinobacteria\" Actinobacteria Actin...    3.0\n",
       "26    MID1  Bacteria \"Actinobacteria\" Actinobacteria Actin...    1.0\n",
       "27    MID1  Bacteria \"Actinobacteria\" Actinobacteria Actin...   99.0\n",
       "28    MID1  Bacteria \"Actinobacteria\" Actinobacteria Actin...   17.0\n",
       "29    MID1  Bacteria \"Actinobacteria\" Actinobacteria Actin...    2.0\n",
       "..     ...                                                ...    ...\n",
       "69    MID8  Bacteria \"Firmicutes\" \"Clostridia\" Clostridial...    6.0\n",
       "70    MID8  Bacteria \"Firmicutes\" \"Clostridia\" Clostridial...   15.0\n",
       "71    MID8  Bacteria \"Firmicutes\" \"Clostridia\" Clostridial...    6.0\n",
       "72    MID8  Bacteria \"Firmicutes\" \"Clostridia\" Clostridial...    1.0\n",
       "73    MID8  Bacteria \"Firmicutes\" \"Clostridia\" Clostridial...  616.0\n",
       "74    MID8  Bacteria \"Firmicutes\" \"Erysipelotrichi\" \"Erysi...    1.0\n",
       "75    MID8  Bacteria \"Fusobacteria\" \"Fusobacteria\" \"Fusoba...    1.0\n",
       "76    MID8  Bacteria \"Nitrospira\" \"Nitrospira\" \"Nitrospira...    2.0\n",
       "77    MID8  Bacteria \"Planctomycetes\" \"Planctomycetacia\" P...    3.0\n",
       "78    MID8  Bacteria \"Proteobacteria\" Alphaproteobacteria ...    1.0\n",
       "79    MID8  Bacteria \"Proteobacteria\" Betaproteobacteria B...    1.0\n",
       "80    MID8  Bacteria \"Proteobacteria\" Betaproteobacteria B...    1.0\n",
       "81    MID8  Bacteria \"Proteobacteria\" Betaproteobacteria H...    3.0\n",
       "82    MID8  Bacteria \"Proteobacteria\" Deltaproteobacteria ...    1.0\n",
       "83    MID8  Bacteria \"Proteobacteria\" Deltaproteobacteria ...    1.0\n",
       "84    MID8  Bacteria \"Proteobacteria\" Deltaproteobacteria ...    1.0\n",
       "85    MID8  Bacteria \"Proteobacteria\" Deltaproteobacteria ...    2.0\n",
       "86    MID8  Bacteria \"Proteobacteria\" Gammaproteobacteria ...   22.0\n",
       "87    MID8  Bacteria \"Proteobacteria\" Gammaproteobacteria ...    1.0\n",
       "88    MID8  Bacteria \"Proteobacteria\" Gammaproteobacteria ...    8.0\n",
       "89    MID8  Bacteria \"Proteobacteria\" Gammaproteobacteria ...    1.0\n",
       "90    MID8  Bacteria \"Proteobacteria\" Gammaproteobacteria ...    1.0\n",
       "91    MID8  Bacteria \"Proteobacteria\" Gammaproteobacteria ...    3.0\n",
       "92    MID8  Bacteria \"Proteobacteria\" Gammaproteobacteria ...    1.0\n",
       "93    MID8  Bacteria \"Proteobacteria\" Gammaproteobacteria ...    3.0\n",
       "94    MID8  Bacteria \"Thermodesulfobacteria\" Thermodesulfo...    1.0\n",
       "95    MID8  Bacteria \"Thermotogae\" Thermotogae Thermotogal...    4.0\n",
       "96    MID8  Bacteria \"Thermotogae\" Thermotogae Thermotogal...    2.0\n",
       "97    MID8  Bacteria Cyanobacteria Cyanobacteria  Chloropl...    1.0\n",
       "98    MID8  Bacteria Cyanobacteria Cyanobacteria  Chloropl...    4.0\n",
       "\n",
       "[2115 rows x 3 columns]"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "import glob as glob\n",
    "import pandas as pd\n",
    "\n",
    "def read_files():  \n",
    "    files = glob.glob(DATA_FOLDER + \"/microbiome/MID*.xls\")\n",
    "    meta = pd.read_excel(DATA_FOLDER + \"/microbiome/metadata.xls\")\n",
    "    df = pd.DataFrame()\n",
    "    df['BARCODE'] = None\n",
    "    i = 0\n",
    "    for file in files:\n",
    "        df = df.append(pd.read_excel(file, header=None))\n",
    "        df.fillna(meta.iloc[i], inplace = True)\n",
    "        i += 1\n",
    "    return df.fillna(\"unknown\")\n",
    "\n",
    "read_files()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 3. Class War in Titanic\n",
    "\n",
    "Use pandas to import the data file `Data/titanic.xls`. It contains data on all the passengers that travelled on the Titanic."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<hr><h2>Data frame:titanic3</h2>1309 observations and 14 variables, maximum # NAs:1188<hr>\n",
       "<TABLE BORDER>\n",
       "<tr><td>Name</td><td>Labels</td><td>Units</td><td>Levels</td><td>Storage</td><td>NAs</td></tr>\n",
       "<tr><td>pclass</td><td></td><td></td><td><a href=\"#pclass\"><div align=right>  3</div></a></td><td>integer</td><td><div align=right>   0</div></td></tr>\n",
       "<tr><td>survived</td><td>Survived</td><td></td><td><div align=right></div></td><td>double</td><td><div align=right>   0</div></td></tr>\n",
       "<tr><td>name</td><td>Name</td><td></td><td><div align=right></div></td><td>character</td><td><div align=right>   0</div></td></tr>\n",
       "<tr><td>sex</td><td></td><td></td><td><a href=\"#sex\"><div align=right>  2</div></a></td><td>integer</td><td><div align=right>   0</div></td></tr>\n",
       "<tr><td>age</td><td>Age</td><td>Year</td><td><div align=right></div></td><td>double</td><td><div align=right> 263</div></td></tr>\n",
       "<tr><td>sibsp</td><td>Number of Siblings/Spouses Aboard</td><td></td><td><div align=right></div></td><td>double</td><td><div align=right>   0</div></td></tr>\n",
       "<tr><td>parch</td><td>Number of Parents/Children Aboard</td><td></td><td><div align=right></div></td><td>double</td><td><div align=right>   0</div></td></tr>\n",
       "<tr><td>ticket</td><td>Ticket Number</td><td></td><td><div align=right></div></td><td>character</td><td><div align=right>   0</div></td></tr>\n",
       "<tr><td>fare</td><td>Passenger Fare</td><td>British Pound (\\243)</td><td><div align=right></div></td><td>double</td><td><div align=right>   1</div></td></tr>\n",
       "<tr><td>cabin</td><td></td><td></td><td><a href=\"#cabin\"><div align=right>187</div></a></td><td>integer</td><td><div align=right>   0</div></td></tr>\n",
       "<tr><td>embarked</td><td></td><td></td><td><a href=\"#embarked\"><div align=right>  3</div></a></td><td>integer</td><td><div align=right>   2</div></td></tr>\n",
       "<tr><td>boat</td><td></td><td></td><td><a href=\"#boat\"><div align=right> 28</div></a></td><td>integer</td><td><div align=right>   0</div></td></tr>\n",
       "<tr><td>body</td><td>Body Identification Number</td><td></td><td><div align=right></div></td><td>double</td><td><div align=right>1188</div></td></tr>\n",
       "<tr><td>home.dest</td><td>Home/Destination</td><td></td><td><div align=right></div></td><td>character</td><td><div align=right>   0</div></td></tr>\n",
       "</TABLE>\n",
       "<hr>\n",
       "<TABLE BORDER>\n",
       "<tr><td>Variable</td><td>Levels</td></tr>\n",
       "<tr><td><a name=\"pclass\">pclass</a></td><td>1st</td></tr>\n",
       "<tr><td></td><td>2nd</td></tr>\n",
       "<tr><td></td><td>3rd</td></tr>\n",
       "<tr><td><a name=\"sex\">sex</a></td><td>female</td></tr>\n",
       "<tr><td></td><td>male</td></tr>\n",
       "<tr><td><a name=\"cabin\">cabin</a></td><td></td></tr>\n",
       "<tr><td></td><td>A10</td></tr>\n",
       "<tr><td></td><td>A11</td></tr>\n",
       "<tr><td></td><td>A14</td></tr>\n",
       "<tr><td></td><td>A16</td></tr>\n",
       "<tr><td></td><td>A18</td></tr>\n",
       "<tr><td></td><td>A19</td></tr>\n",
       "<tr><td></td><td>A20</td></tr>\n",
       "<tr><td></td><td>A21</td></tr>\n",
       "<tr><td></td><td>A23</td></tr>\n",
       "<tr><td></td><td>A24</td></tr>\n",
       "<tr><td></td><td>A26</td></tr>\n",
       "<tr><td></td><td>A29</td></tr>\n",
       "<tr><td></td><td>A31</td></tr>\n",
       "<tr><td></td><td>A32</td></tr>\n",
       "<tr><td></td><td>A34</td></tr>\n",
       "<tr><td></td><td>A36</td></tr>\n",
       "<tr><td></td><td>A5</td></tr>\n",
       "<tr><td></td><td>A6</td></tr>\n",
       "<tr><td></td><td>A7</td></tr>\n",
       "<tr><td></td><td>A9</td></tr>\n",
       "<tr><td></td><td>B10</td></tr>\n",
       "<tr><td></td><td>B101</td></tr>\n",
       "<tr><td></td><td>B102</td></tr>\n",
       "<tr><td></td><td>B11</td></tr>\n",
       "<tr><td></td><td>B18</td></tr>\n",
       "<tr><td></td><td>B19</td></tr>\n",
       "<tr><td></td><td>B20</td></tr>\n",
       "<tr><td></td><td>B22</td></tr>\n",
       "<tr><td></td><td>B24</td></tr>\n",
       "<tr><td></td><td>B26</td></tr>\n",
       "<tr><td></td><td>B28</td></tr>\n",
       "<tr><td></td><td>B3</td></tr>\n",
       "<tr><td></td><td>B30</td></tr>\n",
       "<tr><td></td><td>B35</td></tr>\n",
       "<tr><td></td><td>B36</td></tr>\n",
       "<tr><td></td><td>B37</td></tr>\n",
       "<tr><td></td><td>B38</td></tr>\n",
       "<tr><td></td><td>B39</td></tr>\n",
       "<tr><td></td><td>B4</td></tr>\n",
       "<tr><td></td><td>B41</td></tr>\n",
       "<tr><td></td><td>B42</td></tr>\n",
       "<tr><td></td><td>B45</td></tr>\n",
       "<tr><td></td><td>B49</td></tr>\n",
       "<tr><td></td><td>B5</td></tr>\n",
       "<tr><td></td><td>B50</td></tr>\n",
       "<tr><td></td><td>B51 B53 B55</td></tr>\n",
       "<tr><td></td><td>B52 B54 B56</td></tr>\n",
       "<tr><td></td><td>B57 B59 B63 B66</td></tr>\n",
       "<tr><td></td><td>B58 B60</td></tr>\n",
       "<tr><td></td><td>B61</td></tr>\n",
       "<tr><td></td><td>B69</td></tr>\n",
       "<tr><td></td><td>B71</td></tr>\n",
       "<tr><td></td><td>B73</td></tr>\n",
       "<tr><td></td><td>B77</td></tr>\n",
       "<tr><td></td><td>B78</td></tr>\n",
       "<tr><td></td><td>B79</td></tr>\n",
       "<tr><td></td><td>B80</td></tr>\n",
       "<tr><td></td><td>B82 B84</td></tr>\n",
       "<tr><td></td><td>B86</td></tr>\n",
       "<tr><td></td><td>B94</td></tr>\n",
       "<tr><td></td><td>B96 B98</td></tr>\n",
       "<tr><td></td><td>C101</td></tr>\n",
       "<tr><td></td><td>C103</td></tr>\n",
       "<tr><td></td><td>C104</td></tr>\n",
       "<tr><td></td><td>C105</td></tr>\n",
       "<tr><td></td><td>C106</td></tr>\n",
       "<tr><td></td><td>C110</td></tr>\n",
       "<tr><td></td><td>C111</td></tr>\n",
       "<tr><td></td><td>C116</td></tr>\n",
       "<tr><td></td><td>C118</td></tr>\n",
       "<tr><td></td><td>C123</td></tr>\n",
       "<tr><td></td><td>C124</td></tr>\n",
       "<tr><td></td><td>C125</td></tr>\n",
       "<tr><td></td><td>C126</td></tr>\n",
       "<tr><td></td><td>C128</td></tr>\n",
       "<tr><td></td><td>C130</td></tr>\n",
       "<tr><td></td><td>C132</td></tr>\n",
       "<tr><td></td><td>C148</td></tr>\n",
       "<tr><td></td><td>C2</td></tr>\n",
       "<tr><td></td><td>C22 C26</td></tr>\n",
       "<tr><td></td><td>C23 C25 C27</td></tr>\n",
       "<tr><td></td><td>C28</td></tr>\n",
       "<tr><td></td><td>C30</td></tr>\n",
       "<tr><td></td><td>C31</td></tr>\n",
       "<tr><td></td><td>C32</td></tr>\n",
       "<tr><td></td><td>C39</td></tr>\n",
       "<tr><td></td><td>C45</td></tr>\n",
       "<tr><td></td><td>C46</td></tr>\n",
       "<tr><td></td><td>C47</td></tr>\n",
       "<tr><td></td><td>C49</td></tr>\n",
       "<tr><td></td><td>C50</td></tr>\n",
       "<tr><td></td><td>C51</td></tr>\n",
       "<tr><td></td><td>C52</td></tr>\n",
       "<tr><td></td><td>C53</td></tr>\n",
       "<tr><td></td><td>C54</td></tr>\n",
       "<tr><td></td><td>C55 C57</td></tr>\n",
       "<tr><td></td><td>C6</td></tr>\n",
       "<tr><td></td><td>C62 C64</td></tr>\n",
       "<tr><td></td><td>C65</td></tr>\n",
       "<tr><td></td><td>C68</td></tr>\n",
       "<tr><td></td><td>C7</td></tr>\n",
       "<tr><td></td><td>C70</td></tr>\n",
       "<tr><td></td><td>C78</td></tr>\n",
       "<tr><td></td><td>C80</td></tr>\n",
       "<tr><td></td><td>C82</td></tr>\n",
       "<tr><td></td><td>C83</td></tr>\n",
       "<tr><td></td><td>C85</td></tr>\n",
       "<tr><td></td><td>C86</td></tr>\n",
       "<tr><td></td><td>C87</td></tr>\n",
       "<tr><td></td><td>C89</td></tr>\n",
       "<tr><td></td><td>C90</td></tr>\n",
       "<tr><td></td><td>C91</td></tr>\n",
       "<tr><td></td><td>C92</td></tr>\n",
       "<tr><td></td><td>C93</td></tr>\n",
       "<tr><td></td><td>C95</td></tr>\n",
       "<tr><td></td><td>C97</td></tr>\n",
       "<tr><td></td><td>C99</td></tr>\n",
       "<tr><td></td><td>D</td></tr>\n",
       "<tr><td></td><td>D10 D12</td></tr>\n",
       "<tr><td></td><td>D11</td></tr>\n",
       "<tr><td></td><td>D15</td></tr>\n",
       "<tr><td></td><td>D17</td></tr>\n",
       "<tr><td></td><td>D19</td></tr>\n",
       "<tr><td></td><td>D20</td></tr>\n",
       "<tr><td></td><td>D21</td></tr>\n",
       "<tr><td></td><td>D22</td></tr>\n",
       "<tr><td></td><td>D26</td></tr>\n",
       "<tr><td></td><td>D28</td></tr>\n",
       "<tr><td></td><td>D30</td></tr>\n",
       "<tr><td></td><td>D33</td></tr>\n",
       "<tr><td></td><td>D34</td></tr>\n",
       "<tr><td></td><td>D35</td></tr>\n",
       "<tr><td></td><td>D36</td></tr>\n",
       "<tr><td></td><td>D37</td></tr>\n",
       "<tr><td></td><td>D38</td></tr>\n",
       "<tr><td></td><td>D40</td></tr>\n",
       "<tr><td></td><td>D43</td></tr>\n",
       "<tr><td></td><td>D45</td></tr>\n",
       "<tr><td></td><td>D46</td></tr>\n",
       "<tr><td></td><td>D47</td></tr>\n",
       "<tr><td></td><td>D48</td></tr>\n",
       "<tr><td></td><td>D49</td></tr>\n",
       "<tr><td></td><td>D50</td></tr>\n",
       "<tr><td></td><td>D56</td></tr>\n",
       "<tr><td></td><td>D6</td></tr>\n",
       "<tr><td></td><td>D7</td></tr>\n",
       "<tr><td></td><td>D9</td></tr>\n",
       "<tr><td></td><td>E10</td></tr>\n",
       "<tr><td></td><td>E101</td></tr>\n",
       "<tr><td></td><td>E12</td></tr>\n",
       "<tr><td></td><td>E121</td></tr>\n",
       "<tr><td></td><td>E17</td></tr>\n",
       "<tr><td></td><td>E24</td></tr>\n",
       "<tr><td></td><td>E25</td></tr>\n",
       "<tr><td></td><td>E31</td></tr>\n",
       "<tr><td></td><td>E33</td></tr>\n",
       "<tr><td></td><td>E34</td></tr>\n",
       "<tr><td></td><td>E36</td></tr>\n",
       "<tr><td></td><td>E38</td></tr>\n",
       "<tr><td></td><td>E39 E41</td></tr>\n",
       "<tr><td></td><td>E40</td></tr>\n",
       "<tr><td></td><td>E44</td></tr>\n",
       "<tr><td></td><td>E45</td></tr>\n",
       "<tr><td></td><td>E46</td></tr>\n",
       "<tr><td></td><td>E49</td></tr>\n",
       "<tr><td></td><td>E50</td></tr>\n",
       "<tr><td></td><td>E52</td></tr>\n",
       "<tr><td></td><td>E58</td></tr>\n",
       "<tr><td></td><td>E60</td></tr>\n",
       "<tr><td></td><td>E63</td></tr>\n",
       "<tr><td></td><td>E67</td></tr>\n",
       "<tr><td></td><td>E68</td></tr>\n",
       "<tr><td></td><td>E77</td></tr>\n",
       "<tr><td></td><td>E8</td></tr>\n",
       "<tr><td></td><td>F</td></tr>\n",
       "<tr><td></td><td>F E46</td></tr>\n",
       "<tr><td></td><td>F E57</td></tr>\n",
       "<tr><td></td><td>F E69</td></tr>\n",
       "<tr><td></td><td>F G63</td></tr>\n",
       "<tr><td></td><td>F G73</td></tr>\n",
       "<tr><td></td><td>F2</td></tr>\n",
       "<tr><td></td><td>F33</td></tr>\n",
       "<tr><td></td><td>F38</td></tr>\n",
       "<tr><td></td><td>F4</td></tr>\n",
       "<tr><td></td><td>G6</td></tr>\n",
       "<tr><td></td><td>T</td></tr>\n",
       "<tr><td><a name=\"embarked\">embarked</a></td><td>Cherbourg</td></tr>\n",
       "<tr><td></td><td>Queenstown</td></tr>\n",
       "<tr><td></td><td>Southampton</td></tr>\n",
       "<tr><td><a name=\"boat\">boat</a></td><td></td></tr>\n",
       "<tr><td></td><td>1</td></tr>\n",
       "<tr><td></td><td>10</td></tr>\n",
       "<tr><td></td><td>11</td></tr>\n",
       "<tr><td></td><td>12</td></tr>\n",
       "<tr><td></td><td>13</td></tr>\n",
       "<tr><td></td><td>13 15</td></tr>\n",
       "<tr><td></td><td>13 15 B</td></tr>\n",
       "<tr><td></td><td>14</td></tr>\n",
       "<tr><td></td><td>15</td></tr>\n",
       "<tr><td></td><td>15 16</td></tr>\n",
       "<tr><td></td><td>16</td></tr>\n",
       "<tr><td></td><td>2</td></tr>\n",
       "<tr><td></td><td>3</td></tr>\n",
       "<tr><td></td><td>4</td></tr>\n",
       "<tr><td></td><td>5</td></tr>\n",
       "<tr><td></td><td>5 7</td></tr>\n",
       "<tr><td></td><td>5 9</td></tr>\n",
       "<tr><td></td><td>6</td></tr>\n",
       "<tr><td></td><td>7</td></tr>\n",
       "<tr><td></td><td>8</td></tr>\n",
       "<tr><td></td><td>8 10</td></tr>\n",
       "<tr><td></td><td>9</td></tr>\n",
       "<tr><td></td><td>A</td></tr>\n",
       "<tr><td></td><td>B</td></tr>\n",
       "<tr><td></td><td>C</td></tr>\n",
       "<tr><td></td><td>C D</td></tr>\n",
       "<tr><td></td><td>D</td></tr>\n",
       "</TABLE>\n",
       "<hr>\n"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from IPython.core.display import HTML\n",
    "HTML(filename=DATA_FOLDER+'/titanic.html')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For each of the following questions state clearly your assumptions and discuss your findings:\n",
    "1. Describe the *type* and the *value range* of each attribute. Indicate and transform the attributes that can be `Categorical`. \n",
    "2. Plot histograms for the *travel class*, *embarkation port*, *sex* and *age* attributes. For the latter one, use *discrete decade intervals*. \n",
    "3. Calculate the proportion of passengers by *cabin floor*. Present your results in a *pie chart*.\n",
    "4. For each *travel class*, calculate the proportion of the passengers that survived. Present your results in *pie charts*.\n",
    "5. Calculate the proportion of the passengers that survived by *travel class* and *sex*. Present your results in *a single histogram*.\n",
    "6. Create 2 equally populated *age categories* and calculate survival proportions by *age category*, *travel class* and *sex*. Present your results in a `DataFrame` with unique index."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Write your answer here"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
